kimchi/src/circuits/polynomials/generic.rs

```diff
@@ -104,173 +330,165 @@ impl<F: FftField + SquareRootField> ConstraintSystem<F> {
         generic_zeta: F,
     ) -> Evaluations<F, D<F>> {
         let d1 = self.domain.d1;
-
-        let (scalars, alpha) = Self::gnrc_scalars(alphas, w_zeta, generic_zeta);
-
         let n = d1.size as usize;
 
+        // get scalars
+        let scalars = Self::gnrc_scalars(alphas, w_zeta, generic_zeta);
+
+        //
         let mut res = Evaluations::from_vec_and_domain(vec![F::zero(); n], d1);
 
         let scale = self.coefficients8[0].evals.len() / n;
 
-        // w[0](zeta) * qwm[0] + w[1](zeta) * qwm[1] + w[2](zeta) * qwm[2]
-        self.coefficients8
-            .iter()
-            .zip(scalars[1..].iter())
-            .for_each(|(q, s)| {
-                res.evals.par_iter_mut().enumerate().for_each(|(i, e)| {
-                    *e += *s * q.evals[scale * i];
-                });
+        let coeffs = self.coefficients8.iter();
+        for (scalar, coeff) in scalars.into_iter().zip(coeffs) {
+            res.evals.par_iter_mut().enumerate().for_each(|(i, e)| {
+                *e += scalar * coeff[scale * i];
             });
-
-        // multiplication
-        let s = scalars[0];
-        res.evals.par_iter_mut().enumerate().for_each(|(i, e)| {
-            *e += s * self.coefficients8[MUL_COEFF][scale * i];
-        });
-
-        // constant selector
-        let alpha_generic_zeta = alpha * generic_zeta;
-        res.evals.par_iter_mut().enumerate().for_each(|(i, e)| {
-            *e += alpha_generic_zeta * self.coefficients8[CONSTANT_COEFF][scale * i];
-        });
+        }
 
         // l * qwm[0] + r * qwm[1] + o * qwm[2] + l * r * qmm + qc
         res
     }
 
-    /// function to verify the generic polynomials with a witness
+    /// Function to verify the generic polynomials with a witness.
     pub fn verify_generic(
         &self,
         witness: &[DensePolynomial<F>; COLUMNS],
         public: &DensePolynomial<F>,
     ) -> bool {
-        // multiplication
+        // addition (of left, right, output wires)
         let coefficientsm: [_; COLUMNS] =
             array_init(|i| self.coefficients8[i].clone().interpolate());
-        let multiplication = &(&witness[0] * &witness[1]) * &coefficientsm[MUL_COEFF];
-
-        // addition (of left, right, output wires)
-        let mut wires = DensePolynomial::zero();
-        for (w, q) in witness.iter().zip(coefficientsm.iter()).take(GENERICS) {
-            wires += &(w * q);
+        let mut ff = DensePolynomial::zero();
```

Comment:
This appears not to use `alpha`, and doesn't consider the generic selector. Perhaps it should?

Reply:
thinking more about this, I'm wondering how this can work without considering the generic selector... And actually, we never make use of this function anywhere. I propose that we remove this now that we have inline checks in the prover function for each gate.

Reply:
I'll leave it for now, and I'll add the assert back (not sure why it was removed below in the test). Added the multiplication with the generic selector in case we rely on this in the future (we didn't in the test below as all gates were generic I think). Also no need for alphas as we're just testing for completeness here.